static/js/formatters-internal.js

```diff
@@ -348,22 +376,80 @@ export function image(simpleOrComplexImage = {}, desiredSize = '200x', atLeastAs
 
   const urlWithDesiredDims = urlBeforeDimensions + desiredWidth + 'x' + desiredHeight + extension;
 
-  const dynamicUrl = atLeastAsLarge
+  return atLeastAsLarge
     ? _replaceUrlHost(urlWithDesiredDims, 'dynl.mktgcdn.com')
     : _replaceUrlHost(urlWithDesiredDims, 'dynm.mktgcdn.com');
+}
 
-  return Object.assign(
-    {},
-    image,
-    {
-      url: dynamicUrl.replace('http://', 'https://')
+/**
+ * Given an EU image url, returns the dynamic url.
+ *
+ * @param {Object} image The image object. (e.g.
+ * {
+ *   url: 'https://a.eu.mktgcdn.com/f/0/FLVfkpR1IwpWrWDuyNYCJWVYIDfPO6x1QSztXozMIzo.jpg',
+ *   sourceUrl: 'https://a.mktgcdn.com/p/UN9RPhz0V9D8bNZ3XfNpkGnAk6ikFhVmgvntlBjVyMA/1200x675.jpg',
+ *   width: 1200,
+ *   height: 675,
+ * })
```

Comment:
could the image object ever have a different shape? I noticed the param for the formatter is called `simpleOrComplexImage`, but wasn't sure if the main difference between those is handled in line 304, or if there are other things we need to check for

Reply:
I didn't update the logic there. The only difference between a US and EU image is the format of the url so I don't think we'll need to anything extra. I've checked both a EU and US site, and made sure the images on both of them load correctly.

Reply:
Just double checked. A url would always be there for simple/complex images in US/EU and the format match what we expect in the code